framework/Metrics/DTW.py

```diff
@@ -95,14 +99,14 @@ def __evaluateLocal__(self, x, y, weights = None, axis = 0, **kwargs):
       @ In, kwargs, dictionary of parameters characteristic of each metric
       @ Out, value, float, metric result
     """
-    assert isinstance(x, np.ndarray), "The input data x should be numpy.ndarray"
-    assert isinstance(x, np.ndarray), "The input data y should be numpy.ndarray"
+    assert (isinstance(x, np.ndarray))
+    assert (isinstance(x, np.ndarray))
     tempX = copy.copy(x)
     tempY = copy.copy(y)
     if axis == 0:
-      assert len(x) == len(y), "The first dimension of first input is not the same as the first dimension of second input"
+      assert (len(x) == len(y))
     elif axis == 1:
-      assert x.shape[1] == y.shape[1], "The second dimension of first input is not the same as the second dimension of second input"
```

Comment:
I would add the "error" message as comment for future developers (in case)

Reply:
added

Reply:
the syntax is invalid...everything is failing

Reply:
fixed, I found we should be careful with the parenthesis. If we do not use parenthesis in assert, which should be fine, but if we add parenthesis in assert, we should avoid string message inside parenthesis, which will always be True for the assert evaluation.

Reply:
In our code, in most cases, we do not use parenthesis in assert statement.

Reply:
In my case, I always use parentheses, and never use a string statement, just for comparison.  See examples in the data objects.

Since asserts are exclusively for developers, I will often put any comments that would usually go in a string, in a comment above the assert, assuming the developer will go check the assert.